regression/esbmc-cpp/cpp/ch8_7/test.desc

```diff
@@ -9,4 +9,4 @@
   <item_07_test_importance>High</item_07_test_importance>
   <item_08_execution_type>Automated</item_08_execution_type>
   <item_09_author>Ceteli/UFAM</item_09_author>
-<item_10_mode>CORE</item_10_mode><item_11_issue>CXXThisExpr</item_11_issue></test-case>
+<item_10_mode>KNOWNBUG</item_10_mode><item_11_issue>CXXThisExpr</item_11_issue></test-case>
```

Comment:
why?

Reply:
it probably passed before because the operate= was not generated, that is, the assignment was never performed, we have not been able to handle this situation.

Reply:
Is the test case correct or not? If it's correct and we now give a wrong result, then something is broken in this patch.

Reply:
I did some investigation, as I said above, this error is not caused by this patch, we just made the implicit operator work correctly (there was no assignment behavior before, so the verification passed), I tested this TC locally, and I think it is a wrong TC.

Reply:
It's obviously affected by this patch, otherwise, we wouldn't need to change the verdict.

From the investigation that you said you did, what's the problem then? If the program has a bug, we should send a fixed version with this PR, and change the verdict of this test case to failure.

Reply:
I wrote a suggested demo to restore the situation:

```
class Complex {
public:
    Complex(double real = 0.0, double imaginary = 0.0) 
        : real(real), imaginary(imaginary) {}

    Complex operator+( const Complex & ) const;

private:
    double real;       // real part
    double imaginary;  // imaginary part
}; 


int main()
{
   Complex x;
   Complex y(4.3, 8.2);
   Complex z(3.3, 1.1);

   x = y + z;

   return 0;

}
```

result:
```
ERROR: Unrecognized address_of operand:
address_of
* pointer_obj : address_of
    * pointer_obj : symbol
        * name : c:@F@main#::$tmp::return_value$_operator+$1
        * renamelev : Level 1
        * level1_num : 1
        * level2_num : 0
        * thread_num : 0
        * node_num : 0
        * type : struct
          * members : 

          * member_names : 

          * member_pretty_names : 

          * typename : Complex
          * packed : false
    * type : pointer
      * subtype : struct
          * members : 

          * member_names : 

          * member_pretty_names : 

          * typename : Complex
          * packed : false
* type : pointer
  * subtype : symbol
      * symbol_name : tag-Complex
Aborted
```
Obviously the error is` x = y + z`; 
> there was no assignment behavior before, so the verification passed.